pymeasure/instruments/eurotest/eurotestHPP120256.py

```diff
@@ -0,0 +1,320 @@
+        seconds and raises an exception if the voltage output doesn't reach the voltage setting until the timeout time.
+        :param check_period: voltage output will be measured every check_period (seconds) time
+        :param timeout: time (seconds) give to the voltage output to reach the voltage setting
+        :return: None
+        :raises: Exception if the voltage output can't reach the voltage setting before the timeout (seconds)
+        """
+        logging.info(f"Executing the wait_for_voltage_output_set function.")
+
+        ref_time = time.time()
+        future_time = ref_time + timeout
+
+        voltage_output_setting = self.voltage[1]
+        time.sleep(self.COMMAND_DELAY)
+        error = 0.02  # error of +-ten volts (0.01kV) to enter into the voltage stable window
+        voltage_output = self.measure_voltage[1]
+        voltage_output_set = (voltage_output > (voltage_output_setting - error)) and \
+                             (voltage_output < (voltage_output_setting + error))
+
+        while not voltage_output_set:
+            actual_time = time.time()
+            logging.info(f"\tWaiting for voltage output set. Reading output voltage every {check_period} seconds.\n"
+                         f"\tTimeout: {timeout} seconds. Elapsed time: "
+                         f"{round(actual_time - ref_time, ndigits=1)} seconds.")
+
+            time.sleep(check_period)  # wait for voltage output reaches the voltage output setting
+            voltage_output = self.measure_voltage[1]
+            voltage_output_set = (voltage_output > (voltage_output_setting - error)) and \
+                                 (voltage_output < (voltage_output_setting + error))
+            logging.debug("voltage_output_valid_range: "
+                          "[" + str(voltage_output_setting - error) + ", " + str(voltage_output_setting + error) + "]")
+            logging.debug("voltage_output: " + str(voltage_output))
+            if actual_time > future_time:
+                raise Exception("Timeout for wait_for_voltage_output_set function")
+
+        # if you are here is because the voltage_setting has been reaches at the output of the HV Voltage source
+
+    # Constructor
+    def __init__(self, adapter, response_encoding="iso-8859-2", query_delay=0.1, **kwargs):
+        self.response_encoding = response_encoding  # This instrument use this encoding, so we have to take into account
+        self.query_delay = query_delay  # Delay in s to sleep between the write and read occuring in a query
+        adapter.connection.encoding = self.response_encoding  # comment it if you want to run the test
+        adapter.connection.query_delay = self.query_delay  # comment it if you want to run the test
+        super().__init__(
+            adapter,
+            "Euro Test High Voltage DC Source model HPP-120-256",
+            **kwargs
+        )
```

Comment:
By convention this should be the first entry after the doc-string in the class.

Reply:
Yes. But I wanted to modify the adapter before passing it to the constructor of the Superclass. What do you suggest I could resolve it?
Many thanks.

Reply:
As mentioned [here](https://github.com/pymeasure/pymeasure/pull/701#discussion_r991053525), unless you have major work to do with the adapter, I would just put the termination settings as defaults in the  `___init___`block:
```
    def __init__(self, adapter, **kwargs):
        kwargs.setdefault('read_termination', '\r\n')
        kwargs.setdefault('write_termination', '')
        kwargs.setdefault('send_end', True)
        super().__init__(
            adapter,
            "Euro Test High Voltage DC Source model HPP-120-256",
            **kwargs,
        )
```

similiar could be done for querydelay etc.

Reply:
Yes. But it doesn't work for querydelay and encoding. 

For example, next code doesn't work for enconding:

    def __init__(self, adapter, response_encoding="iso-8859-2", query_delay=0.1, **kwargs):
        # This instrument use encoding iso-8859-2 on response, so we have to take it into account        
        kwargs.setdefault('encoding', response_encoding)
        super().__init__(
            adapter,
            "Euro Test High Voltage DC Source model HPP-120-256",
            **kwargs
        )

But this other works:

    def __init__(self, adapter, response_encoding="iso-8859-2", query_delay=0.1, **kwargs):

        # This instrument uses iso-8859-2 encoding, so we have to take it into account
        # Also query_delay has to be taken into account for slow response instruments
        self.response_encoding = response_encoding
        self.query_delay = query_delay
        adapter.connection.encoding = self.response_encoding
        adapter.connection.query_delay = self.query_delay
        super().__init__(
            adapter,
            "Euro Test High Voltage DC Source model HPP-120-256",
            **kwargs
        )

Any suggestion will be apreciate. Thanks.

Reply:
According to the pyvisa documentation and source code, the kwargs are executed afterwards and just do, what you did: setting the encoding and query_delay.

You could add it **after** the super.init:
```
super().__init__(...)
if isinstance(adapter, VISAAdapter):
    self.adapter.connection.encoding = xy
    self.adapter.connection.query_delay = query_delay
```

The if question is necessary, as other adapters do not have an encoding or query_delay.

Reply:
I think now the code follow your recomendations. Many thanks.